### PR TITLE
General fixes & polishing

### PR DESCRIPTION
**Pull request by vesln**
## Scope

- Make the vote button in the default tab look like it cannot be clicked (lukas)
- Remove the fade in effect in the default tab (rado)
- The header link now points to "http" not "https" (ryan)
- Fix the shadow of the Product Pane (ryan)
- Make the product pane wider (ryan)
- Remove padding/margin from the popup window (ryan)
- Replace FB like with FB share (ryan)
- Fix the styling of the close button (ryan)
- Make the PH widget wider (ryan)
- Show loading indicator when opening the product details (andreas)

## Notes

None

### CODE REVIEW

File: src/apps/background/main.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
-// TODO(vesln): Implement suggestions in the future?
-
 require('./x-frame');
 
 /**
+ * Dependencies.
+ */
+
+let analytis = require('../../common/analytics');
```
Comment (andreasklinger) [inline src/apps/background/main.js]: typo

Comment (ayrton) [inline src/apps/background/main.js]: typo `analytis` vs `analytics`

File: src/apps/content/main.js [hunk 2]
```diff
@@ -13,33 +13,51 @@ let getHost = require('./util/get-host');
...
+    api.getProduct(location.href);
+  } else {
+    debug('not showing product bar...');
+  }
 }
+
+let option = {};
+option[BAR_DISABLED_KEY] = false;
+
+chrome.storage.sync.get(option, function(items) {
```
Comment (andreasklinger) [inline src/apps/content/main.js]: option*s* ?

File: src/apps/content/util/detector.js [hunk 3]
```diff
@@ -7,43 +19,15 @@
...
-    if (this.host === this.currentHost) return false;
-
-    // different referrer
-    if (this.referrer !== this.host) return false;
-
-    // all good
-    return true;
+  enable(search) {
+    return qs.parse(search).ref === REF;
   }
```
Comment (andreasklinger) [inline src/apps/content/util/detector.js]: <3

File: src/common/analytics/index.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ * Generate random user id.
+ *
+ * @returns {String}
+ * @private
+ */
+
+function anonymousId() {
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+    var r = Math.random()*16|0, v = c == 'x' ? r : (r&0x3|0x8);
+    return v.toString(16);
```
Comment (andreasklinger) [inline src/common/analytics/index.js]: Shouldnt you store this in localStorage?
or how do you know next time it's the same person?

File: src/apps/content/main.js [hunk 5]
```diff
@@ -13,33 +13,51 @@ let getHost = require('./util/get-host');
...
+
+    // fetch the product data
+    api.getProduct(location.href);
+  } else {
+    debug('not showing product bar...');
+  }
 }
+
+let option = {};
+option[BAR_DISABLED_KEY] = false;
```
Comment (RStankov) [inline src/apps/content/main.js]: You can have:

```
{
   [BAR_DISABLED_KEY]: false
}
```

File: src/apps/tabs/main.js [hunk 6]
```diff
@@ -10,11 +12,49 @@ let React = require('react');
...
+}
+
+let option = {};
+option[TAB_DISABLED_KEY] = false;
 
-React.render(
-  <DefaultTab />,
-  containerEl
-);
+chrome.storage.sync.get(option, function(items) {
```
Comment (RStankov) [inline src/apps/tabs/main.js]: this `chrome.storage.sync` is used in several places. I think you can have a module just name `Cache` or similar.

Comment (RStankov) [inline src/apps/tabs/main.js]: Actually when I had a second look the whole structure can be like:

```js
function(name, render) {
  let option = {};
  option[name] = false;

  chrome.storage.sync.get(option, function(items) {
    if (!items[name]) {
      render();
    } else {
      debug('tab disabled');
   }
 });
}
```

File: src/apps/popup/components/popup.react.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    return { url: SEARCH_URL };
+  },
+
+  /**
+   * Render the view.
+   */
+
+  render() {
+    let postClass = this.state.url === POST_URL ? 'current': null;
+    let searchClass = this.state.url === SEARCH_URL ? 'current': null;
```
Comment (RStankov) [inline src/apps/popup/components/popup.react.js]: Are you using those?

File: src/apps/options/main.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Dependencies.
+ */
+
+let Options = require('./components/options.react');
+
+// render the options page
+let containerEl = document.createElement('div');
+document.body.insertBefore(containerEl, document.body.firstChild)
+React.render(<Options />, containerEl);
```
Comment (RStankov) [inline src/apps/options/main.js]: I see this pattern in several places:

```
function reactReact(components) {
  let containerEl = document.createElement('div');
  document.body.insertBefore(containerEl, document.body.firstChild)
  React.render(components, containerEl);
}
```

File: src/common/analytics/index.js (hunk 4 above)
Comment (vesln) [inline src/common/analytics/index.js]: Yep, we are storing it in chrome storage

**Pull request merged**